modules/storage/storage.go

```diff
@@ -167,17 +167,11 @@ func (s *VolumeSource) ToCoreVolumeSource() (*corev1.VolumeSource, error) {
 
 // ToCoreVolume - convert Volume to corev1.Volume
 func (s *Volume) ToCoreVolume() (*corev1.Volume, error) {
-	coreVolume := &corev1.Volume{}
-
-	coreVolumeBytes, err := json.Marshal(s)
-	if err != nil {
-		return nil, fmt.Errorf("error marshalling Volume: %w", err)
-	}
-
-	err = json.Unmarshal(coreVolumeBytes, coreVolume)
-	if err != nil {
-		return nil, fmt.Errorf("error unmarshalling Volume: %w", err)
+	volSource, err := s.VolumeSource.ToCoreVolumeSource()
+	coreVolume := &corev1.Volume{
+		Name:         s.Name,
+		VolumeSource: *volSource,
```

Comment:
in case of error when *volSource is nil this will hit a nil ptr dereference error. should check for err condition before.

Reply:
good call, pushed a fix for that